### PR TITLE
screenshots in system tests

### PR DESCRIPTION
**Pull request by jywarren**
Re #5320 but squashed!

### CODE REVIEW

File: containers/docker-compose-testing.yml [hunk 1]
```diff
@@ -27,6 +27,22 @@ services:
       - RAILS_ENV=${RAILS_ENV}
       - SECRET_KEY_BASE=${SECRET_KEY_BASE}
       - REDIS_URL=redis://redis:6379/0
+      - DANGER_GITHUB_API_TOKEN=${DANGER_GITHUB_API_TOKEN}
+      - TRAVIS_PULL_REQUEST=${TRAVIS_PULL_REQUEST}
+      - TRAVIS_REPO_SLUG=${TRAVIS_REPO_SLUG}
+      - CI=${CI}
+      - TRAVIS=${TRAVIS}
+      - CONTINUOUS_INTEGRATION=${CONTINUOUS_INTEGRATION}
+      - HAS_JOSH_K_SEAL_OF_APPROVAL=${HAS_JOSH_K_SEAL_OF_APPROVAL}
```
Comment (jywarren) [inline containers/docker-compose-testing.yml]: Was this just for demonstration purposes, @icarito ?

**Comment by jywarren** [comment]
Ohhhhh so this will only work on PRs on publiclab's account?

File: containers/docker-compose-testing.yml (hunk 1 above)
Comment (icarito) [reply containers/docker-compose-testing.yml]: I don't remember off the top of my head why HAS_JOSH_K_SEAL_OF_APPROVAL is needed but it is no invention of mine and it seems some libs check for it!

Comment (icarito) [reply containers/docker-compose-testing.yml]: https://docs.travis-ci.com/user/environment-variables/

**Comment by jywarren** [comment]
Ok, so my fork doesn't have access to `publiclab`'s environment variables. 

https://docs.travis-ci.com/user/environment-variables/#encrypting-environment-variables

**Comment by jywarren** [comment]
> Signed Policy Documents: Specify what can be uploaded to a bucket. Policy documents allow greater control over size, content type, and other upload characteristics than signed URLs, and can be used by website owners to allow visitors to upload files to Cloud Storage.

https://cloud.google.com/storage/docs/access-control/

https://cloud.google.com/storage/docs/xml-api/post-object#policydocument

**Comment by jywarren** [comment]
https://stackoverflow.com/questions/25158266/how-to-allow-anonymous-uploads-to-cloud-storage

**Comment by jywarren** [comment]
Huh... 

```
[Screenshot]: tmp/screenshots/failures_test_posting_from_the_editor.png
ERROR["test_posting_from_the_editor", #<Minitest::Reporters::Suite:0x000055c95609e8e8 @name="PostTest">, 74.22745889599992]
 test_posting_from_the_editor#PostTest (74.23s)
Capybara::ElementNotFound:         Capybara::ElementNotFound: Unable to find field "Title" that is not disabled
            test/system/post_test.rb:17:in `block in <class:PostTest>'
```

Could be timing related... I rebooted system tests!

**Comment by jywarren** [comment]
Whoa, this passed tests and all, but no uploads?

**Comment by jywarren** [comment]
Hmm...

```


1.40s$ if [ $TRAVIS_PULL_REQUEST != "false" ]; then docker-compose exec web danger --verbose; fi
WARNING: The SECRET_KEY_BASE variable is not set. Defaulting to a blank string.
WARNING: The GITHUB_ACTION variable is not set. Defaulting to a blank string.
WARNING: The GITHUB_TOKEN variable is not set. Defaulting to a blank string.
WARNING: The GITHUB_REPOSITORY variable is not set. Defaulting to a blank string.
WARNING: The GITHUB_EVENT_PATH variable is not set. Defaulting to a blank string.
WARNING: The GITHUB_EVENT_NAME variable is not set. Defaulting to a blank string.
Not a GitHubActions Pull Request - skipping `danger` run.
The command "if [ $TRAVIS_PULL_REQUEST != "false" ]; then docker-compose exec web danger --verbose; fi" exited with 0.
```

**Comment by jywarren** [comment]
Uh, ok - https://github.com/symfony/panther/issues/66#issuecomment-407826412, https://blog.travis-ci.com/august-2012-upcoming-ci-environment-updates/

`HAS_JOSH_K_SEAL_OF_APPROVAL` is necessary to show we are in Travis environment. Weird.

**Comment by jywarren** [comment]
I changed it so if it can't find the JUnit reports, it just warns, doesn't fail.

**Pull request merged**